extendr-api/src/wrapper/integers.rs

```diff
@@ -98,4 +106,57 @@ impl Integers {
     pub fn no_na(&self) -> Bool {
         unsafe { INTEGER_NO_NA(self.get()).into() }
     }
+
+    /// Return an iterator for a non-altrep object.
+    /// Suitable for short vectors.
+    /// ```
+    /// use extendr_api::prelude::*;
+    /// test! {
+    ///     let vec = Integers::from_values(0..3);
+    ///     assert_eq!(vec.iter().cloned().sum::<i32>(), 3);
+    /// }
+    /// ```
+    pub fn iter(&self) -> impl Iterator<Item = &i32> {
```

Comment:
Apologies for being nitpicky: Shouldn't the item type be `Rint`? In the general case, we don't know that we can return an `i32` for sure (could be `NA`).

Reply:
I thought this would return `SliceIter<T>`. Or, is the idea that doing `NA`-validation beforehand at once, which is similar to this comment above? If so, `i32` might make sense.

> do all its checks once for a number of elements

Reply:
Good point...

Reply:
I've implemented Rint where possible.

I'm in two minds about get_region as we would like to support SIMD arithmetic but can do this too if required.

Reply:
The SliceIter should probably go. Its only useful feature is that the lifetime
is effectively static.